commons/src/store/mailbox.ts

```diff
@@ -248,10 +248,71 @@ function initializeThreads() {
             return { ...threads };
           });
         }
+
+        //Remove existing draft in thread if sent as draft
+        update((threads) => {
+          let threadToUpdate = threads[queryKey][currentPage].threads.find(
+            (thread) => thread.id === foundThread.id,
+          );
+          if (threadToUpdate) {
+            const filteredDrafts = foundThread.drafts.filter(
+              (draft) => draft.id !== incomingMessage.id,
+            );
+            foundThread.drafts = filteredDrafts;
```

Comment:
You can cut down on the duplicate code here by adding an additional param of `messageType` ([enum here](https://github.com/nylas/components/blob/main/commons/src/enums/Nylas.ts#L19-L22)) to `hydrateThreadInMessage`, and adding these lines to [this block](https://github.com/nylas/components/blob/5f5b2dbc7f3219dbacc1bf3ccd4e17c30129cc05/commons/src/store/mailbox.ts#L225-L227) and check if `messageType === MessageType.DRAFTS`.

```
  hydrateMessageInThread: (
    incomingMessage: Message,
    query: MailboxQuery,
    currentPage: number,
    messageType: string = "messages"
  ) 
```

```
if (threadToUpdate) {
  if (messageType === MessageType.DRAFTS) {
    const filteredDrafts = foundThread.drafts.filter(
    (draft) => draft.id !== incomingMessage.id,
  );
  threadToUpdate = JSON.parse(JSON.stringify(foundThread));
}
```